### PR TITLE
workaround for dasvader HDF2 compatibility issue.

### PR DESCRIPTION
**Pull request by d-chambers**
• ## Description

  This PR makes DASVader legacy JLD2 handling explicit on newer HDF5 stacks.

  Legacy DASVader files can contain anonymous JLD2 object references which read successfully on older stacks but fail on newer h5py/HDF5 combinations. Instead of letting those failures surface as low-level h5py errors, this PR adds a dedicated compatibility error path and treats it as a dependency/compatibility issue in tests and benchmarks.

  Changes included:

  - add a shared DependencyError base in dascore.exceptions
  - add DASVaderCompatibilityError for legacy DASVader files that require an older HDF5 stack
  - detect anonymous DASVader JLD2 refs in the reader and raise a clear compatibility message with install guidance
  - document the compatibility note in the DASVader reader docstring
  - add a synthetic modern DASVader fixture with named references so DASVader read/scan/common-IO coverage still runs on newer stacks
  - update DASVader tests to skip on compatibility-stack errors instead of failing
  - simplify benchmark exception handling to suppress DependencyError, which now covers both missing optional deps and DASVader compatibility issues

  ## Checklist

  I have (if applicable):

  - [ ] referenced the GitHub issue this PR closes.
  - [x] documented the new feature with docstrings and/or appropriate doc page.
  - [x] included tests. See testing guidelines (https://dascore.org/contributing/testing.html).
  - [ ] added the "ready_for_review" tag once the PR is ready to be reviewed.

## Release Notes

* **New Features**
  * Enhanced error handling and reporting for dependency-related compatibility issues

* **Bug Fixes**
  * Improved detection of legacy file format incompatibilities with more informative error messages

* **Tests**
  * Expanded test coverage to verify handling of both modern and legacy file formats, including edge cases like out-of-range read operations

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 from contextlib import suppress
 from pathlib import Path
 
+import h5py
```
[the remark this answers is not in the document]
Comment (d-chambers) [reply tests/conftest.py]: h5py is a core dep.

File: tests/test_io/test_dasvader/test_dasvader.py [hunk 2]
```diff
@@ -60,15 +65,52 @@ def das_vader_strainrate_no_attrib_path(self, tmp_path_factory):
...
+        assert len(scanned) == 1
+        attrs = scanned[0]
+        assert attrs.file_format == "DASVader"
+        assert attrs.gauge_length == 10.0
+        assert attrs.host_name == "test-host"
+        patch = dc.read(
+            dasvader_modern_path, time=(attrs.time_min + dc.to_timedelta64(1), ...)
+        )[0]
+        assert patch.dims == ("distance", "time")
+        assert patch.attrs.gauge_length == 10.0
```
Comment (d-chambers) [inline tests/test_io/test_dasvader/test_dasvader.py]: Can we not hardcode these values? Maybe a common class or namespace for the fixture and test assertions?

File: tests/test_io/test_dasvader/test_dasvader.py [hunk 3]
```diff
@@ -60,15 +65,52 @@ def das_vader_strainrate_no_attrib_path(self, tmp_path_factory):
...
     def test_strainrate_without_atrib(self, das_vader_strainrate_no_attrib_path):
         """Ensure parsing works when data is `strainrate` and `atrib` is absent."""
         patch = dc.read(das_vader_strainrate_no_attrib_path)[0]
         assert patch.dims == ("time", "distance")
         assert "gauge_length" not in patch.attrs.model_dump()
         scanned = dc.scan(das_vader_strainrate_no_attrib_path)
         assert len(scanned) == 1
+
+    def test_julia_ms_to_datetime64_unwraps_nested_void(self):
+        """Ensure nested JLD2-style void values are unwrapped."""
```
Comment (d-chambers) [inline tests/test_io/test_dasvader/test_dasvader.py]: Elaborate more on these voids, why they are used and their semantic meaning.

File: tests/test_io/test_common_io.py [hunk 4]
```diff
@@ -156,6 +156,8 @@ def io_path_tuple(request):
     This is used for common testing.
     """
     io, fetch_name = request.param
+    if isinstance(io, DASVaderV1) and fetch_name == "das_vader_1.jld2":
+        return io, request.getfixturevalue("dasvader_modern_path")
```
Comment (d-chambers) [inline tests/test_io/test_common_io.py]: Let's just remove dasvader from the common paths, just put it in its own tests in the dasvader test module.

File: tests/test_io/test_common_io.py [hunk 5]
```diff
@@ -164,6 +166,8 @@ def io_path_tuple(request):
 def data_file_path(request):
     """A fixture of all data files. Will download if needed."""
     param = request.param
+    if str(param) == "das_vader_1.jld2":
+        return request.getfixturevalue("dasvader_modern_path")
```
Comment (d-chambers) [inline tests/test_io/test_common_io.py]: Same here; remove this ugly logic in favor of dasvader specific tests.

File: tests/conftest.py [hunk 6]
```diff
@@ -251,6 +252,81 @@ def brady_hs_das_dts_coords_path():
...
+        fi.create_dataset("dDAS", data=ddas)
+    return path
+
+
+@pytest.fixture(scope="session")
+def dasvader_modern_path(tmp_path_factory):
+    """Create a readable DASVader file with named datasets."""
+    path = tmp_path_factory.mktemp("dasvader_modern") / "modern_named_refs.jld2"
+    return _write_dasvader_file(path)
+
```
Comment (d-chambers) [inline tests/conftest.py]: We can move this to the dasvader test module.
